public/js/build/BuildRecipeFunctionFactory.js

```diff
@@ -8,51 +8,100 @@ class BuildRecipeFunctionFactory {
+        ff.createNotification("success", ingredient_name);
       }
     };
   }
 
   /*
-  The createRemoveIngredientFunction function takes a table row to remove and an ingredient's ID
-  as the first two parameters.
-  The third parameter is an instance of the BuildRecipeViewController class.
-  It returns a function that will remove the ingredient from recipe-table-body and the recipe_ingredients object.
-   */
-  createRemoveIngredientFunction(row_to_remove, ingredient_id, brvc) {
-    return function () {
-      let recipe_table_body = row_to_remove.parentElement;
-      recipe_table_body.removeChild(row_to_remove);
-      delete brvc.recipe_ingredients[ingredient_id];
+  The createNotification function takes two strings as parameters: the notification type and ingredient name.
+  It creates a visual indicator when the user makes changes to the recipe.
+  */
+  createNotification(alertType, ingredient_name) {
+    let alertDiv = document.getElementById("alert-div");
+    let alert = document.createElement("div");
+    alert.classList.add("alert");
+    let alertString = `${ingredient_name} was added to the recipe.`;
+    
+    // modify notification appearance based on alertType
+    if (alertType == "error") {
+      alertString = `${ingredient_name} is already in the recipe.`;
+      alert.classList.add("add-error")
+    } else if (alertType == "remove") {
+      alertString = `${ingredient_name} was removed from the recipe.`;
+      alert.classList.add("remove-success")
     }
+
+    alert.textContent = alertString;
+    alertDiv.prepend(alert);
+
+    setTimeout(() => {
+      alert.remove()
+    }, 8000);
+  }
+
+  /* The createIdCell function returns a table cell containing the given ingredient's ID. */
+  createIdCell(ingredient_id) {
+    let id_cell = document.createElement('td');
+    id_cell.textContent = ingredient_id;
+    return id_cell;
+  }
+
+  /* The createNameCell function returns a table cell containing the given ingredient's name. */
+  createNameCell(ingredient_name) {
+    let name_cell = document.createElement('td');
+    name_cell.textContent = ingredient_name;
+    return name_cell;
+  }
+
+  /* The createRemoveButtonCell function returns a table cell containing a given button. */
+  createButtonCell(button) {
+    let buttonCell = document.createElement('td');
+    buttonCell.appendChild(button);
+    return buttonCell;
+  }
+
+  /* The createRemoveButton function returns a button used to remove the ingredient from the recipe. */
+  createRemoveButton(ingredient_id) {
+    let removeButton = document.createElement('button');
+    removeButton.textContent = "Remove";
+    removeButton.setAttribute("id", "remove-" + ingredient_id);
+    return removeButton;
+  }
+
+  /* The createInfoButton function returns a button used to get ingredient info. */
+  createInfoButton() {
+    let infoButton = document.createElement("button");
+    infoButton.textContent = "Info";
+    return infoButton;
   }
```

Comment:
Similarly, you are inflating the code of `BuildRecipeFunctionFactory` by adding yet another responsibility - managing tables. I have refactored this section of code in my PR by extracting the `TableBuilder` class.
https://github.com/anthonyliuhao/cs361_group_project/pull/30/files